tsql/TSqlParser.g4

```diff
@@ -205,54 +205,108 @@ another_statement
 //https://docs.microsoft.com/en-us/sql/t-sql/statements/alter-application-role-transact-sql
 
 alter_application_role
-    : ALTER APPLICATION ROLE application_role=id  alter_application_role_with_clause
+    : alter_application_role_start application_role=id  alter_application_role_with_clause
+    ;
+
+alter_application_role_start
+    : ALTER APPLICATION ROLE
     ;
 
 alter_application_role_with_clause
-    :WITH
-       (NAME EQUAL new_application_role_name=id COMMA?)
-       (PASSWORD EQUAL  application_role_password=STRING COMMA?)?
-       (DEFAULT_SCHEMA  EQUAL app_role_default_schema=id COMMA?)?
+    : alter_application_role_with
+       (alter_application_role_with_name new_application_role_name=id COMMA?)
+       (alter_application_role_with_password  application_role_password=STRING COMMA?)?
+       (alter_application_role_with_default app_role_default_schema=id COMMA?)?
+    ;
+
+alter_application_role_with
+    : WITH
```

Comment:
Please do not use parse rules for single tokens.

Remove
```ANTLR
alter_application_role_with
    : WITH
    ;
```

And paste `WITH` keyword (ant other) to `alter_application_role_with_clause`:
```ANTLR
 alter_application_role_with_clause
     : WITH
       (NAME EQUAL new_application_role_name=id COMMA?)
       (PASSWORD EQUAL  application_role_password=STRING COMMA?)?
       (DEFAULT_SCHEMA EQUAL app_role_default_schema=id COMMA?)?
    ;
```

This improves readability and simplifies grammar.
This note related to other your changes too.